csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -15,30 +15,66 @@ namespace nvfuser::preseg_passes {
 
 namespace {
 
+// returns allocation order to propagate adjusted for `tv`
+// Right now it merely removes all the entry corresponding to reduction iter
+// domains in tv,
+AllocationOrder adjustAllocationOrder(
+    const TensorView* tv,
+    const AllocationOrder& alloc_order) {
+  AllocationOrder ret = alloc_order;
+
+  int64_t tv_rank = static_cast<int64_t>(tv->nDims());
```

Comment:
```suggestion
  auto tv_rank = static_cast<int64_t>(tv->nDims());
```

Nit: `static_cast<int64_t>` makes the explicit type redundant. Using auto here is less error prone -- people may forget to change the receiver type when they change static_cast.